casepro/backend/junebug.py

```diff
@@ -4,16 +4,54 @@
 from . import BaseBackend
 
 
+class IdentityStore(object):
+    '''Implements required methods for accessing the identity data in the
+    identity store.'''
+    def __init__(self, base_url, auth_token, address_type):
+        '''
+        base_url: the base URL where the identity store is located
+        auth_token: the token that should be used for authorized access
+        address_type: the address type of the addresses that we want
+        '''
+        self.base_url = base_url.rstrip('/')
+        self.address_type = address_type
+        self.session = requests.Session()
+        self.session.headers.update({'Authorization': 'Token %s' % auth_token})
+        self.session.headers.update({'Content-Type': 'application/json'})
+
+    def get_paginated_response(self, url, params={}, **kwargs):
+        '''Get the results of all pages of a response.'''
+        results = []
+        while url is not None:
+            r = self.session.get(url, params=params, **kwargs)
+            data = r.json()
+            results += data.get('results', [])
+            url = data.get('next', None)
+            # params are included in the next url
+            params = {}
+        return results
```

Comment:
Ah, I see that the thing we are using it for is for getting addresses for a particular identity, which probably won't be that many results. We do run the risk of using re-using this method in a situation where there are potentially many results though.

I'm still thinking we should use a generator or something here and maybe for `get_addresses` we could just do a `list(get_paginated_response(...))` since we're not worried about the number of results there?